File: Java/Language/InterfaceTest1/src/Banker.java
package edu.met.banking;

//factory class for Account
public class Banker {

	private static long nid;

	//static initializer block is executed once after the class is
	//loaded by the JVM, used for initializing static fields
	static {
		nid = System.currentTimeMillis() % 1000000;
	}

	public static Account openCurrentAccount() {
		var acc = new CurrentAccount();
		acc.id = ++nid + 100000000;
		return acc;
	}

	public static Account openSavingsAccount() {
		var acc = new SavingsAccount();
		acc.id = ++nid + 200000000;
		return acc;
	}

	//a class which only defines static members does need instances
	//an such it should define a private constructor
	private Banker() {}
}

File: dotnet/Language/Inheritance/AbstractClassTest/BankLib/Banker.cs
namespace Met.Banking;

//a class defined with static modifier can only contain
//static members and as such it cannot be activated nor
//can be used as a base-class
public static class Banker
{
    private static long nid;

    static Banker()
    {
        nid = DateTime.Now.Ticks % 1000000L;
    }

    public static Account OpenCurrentAccount()
    {
        var acc = new CurrentAccount();
        acc.Id = ++nid + 100000000L;
        return acc;
    }

    public static Account OpenSavingsAccount()
    {
        SavingsAccount acc = new();
        acc.Id = ++nid + 200000000L;
        return acc;
    }

    public static void FreezeAccount(Account acc)
    {
        acc.Id = -acc.Id;
    }

}
